### DISCUSSION TITLE
Using BroadcastsEvents with a landlord model

### DISCUSSION THREAD

**Discussion by IronSinew** [spatie/laravel-multitenancy#566] [Q&A] [answered]
When using the `BroadcastsEvents`  trait with a model that has the `UsesLandlordConnection`  trait as well, whenever a change is attempted to broadcast, I get the error:

```
The current tenant could not be determined in a job named `Illuminate\Queue\CallQueuedHandler@call`. No `tenantId` was set in the payload.
```

Is there another way to apply the `NotTenantAware` interface with those jobs being generated by the model changes? It feels super heavy-handed (and I don't know if it will have any other repercussions to add `\Illuminate\Database\Eloquent\BroadcastableModelEventOccurred::class,` to the `not_tenant_aware_jobs` in the `config/multitenancy.php` config file, especially because I'm sure that there will be other model events I'll want to broadcast that will be in tenant-scoped models.

TLDR: Landlord model events being broadcast expect tenant scoping? How can I prevent that?

**Comment by IronSinew** [comment, marked as answer]
I ended up doing this:


```php
<?php

namespace App\Services\Tenancy;

use Illuminate\Database\Eloquent\BroadcastableModelEventOccurred;
use Spatie\Multitenancy\Jobs\NotTenantAware;

class LandlordBroadcastableModelEvents extends BroadcastableModelEventOccurred implements NotTenantAware {}

```

Any landlord model that needs to broadcast:

```php
    protected function newBroadcastableEvent(string $event): BroadcastableModelEventOccurred
    {
        return new LandlordBroadcastableModelEvents(
            $this, $event
        );
    }
```
